src/main/java/slimeknights/tconstruct/smeltery/block/entity/module/MeltingModule.java

```diff
@@ -78,9 +78,9 @@ private void resetRecipe() {
    */
   public void setStack(ItemStack newStack) {
     // send a slot update to the client when items change, so we can update the TESR
-    Level world = parent.getLevel();
+    Level world = parent.getParent().getLevel();
```

Comment:
This looks messy enough that it might be worth keeping "both parents". I would probably call one of them `be` and the other `parent`, or alternatively `be` and `inventory`.